pkg/controllers/clusteroperator_controller.go

```diff
@@ -251,7 +256,55 @@ func (r *CloudOperatorReconciler) provisioningAllowed(ctx context.Context, infra
 		return false, nil
 	}
 
-	// TODO: Add part confirming migration is finished for KCM before allowing provisioning
+	ownedByKCM, err := r.isCloudControllersOwnedByKCM(ctx)
+	if err != nil {
+		return false, err
+	}
+	if ownedByKCM {
+		// KCM resource found and it owns Cloud provider
+		klog.Infof("Kube Controller Manager still owns Cloud Controllers. Skipping...")
+
+		if err := r.setStatusAvailable(ctx); err != nil {
+			klog.Errorf("Unable to sync cluster operator status: %s", err)
+			return false, err
+		}
+
+		return false, nil
+	}
 
 	return true, nil
 }
+
+func (r *CloudOperatorReconciler) isCloudControllersOwnedByKCM(ctx context.Context) (bool, error) {
+	kcm := &operatorv1.KubeControllerManager{}
+	err := r.Get(ctx, client.ObjectKey{Name: kcmResourceName}, kcm)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// If KCM object doesn't exist, we assume that we are in bootstrapping process and
+			// the controllers are not owned by KCM.
+			klog.Info("Kube Config Manager object not found. It's considered that KCM does not own the controllers.")
+			return false, nil
+		}
```

Comment:
I don't think this will work

The object you are fetching (https://github.com/openshift/cluster-kube-controller-manager-operator/blob/e52078fbafd5bc04210185ce546c89cbc83de9b2/manifests/0000_25_kube-controller-manager-operator_01_operator.cr.yaml#L1) is created by CVO before the CCCMO is created.

In this case, I think what we need to do is check whether the status is empty. CVO won't create a status, only KCMO will do that, does that make more sense as a check? Perhaps instead we should make sure status is empty and there's no observed config? 

Could this lead to a race condition during startup? if CCCMO starts before KCMO but isn't supposed to? Perhaps we need to check for KCMO's pod to be pending to be very certain?